llvm/lib/IR/Globals.cpp

```diff
@@ -263,6 +265,16 @@ void GlobalObject::setSection(StringRef S) {
   setGlobalObjectFlag(HasSectionHashEntryBit, !S.empty());
 }
 
+void GlobalObject::setCodeModel(CodeModel::Model M) {
+  getContext().pImpl->GlobalObjectCodeModels[this] = M;
```

Comment:
we have enough bits to store the code model directly in the SubclassData, let's do that for now for simplicity. we can add a note that if we ever run out of bits we can do this way of storing the actual value in LLVMContextImpl